hashin.py

```diff
@@ -175,17 +182,25 @@ def run_single_package(
             new_lines += ' \\'
         new_lines += '\n'
 
-    if verbose:
-        _verbose('Editing', file)
     with open(file) as f:
         requirements = f.read()
     requirements = amend_requirements_content(
         requirements,
         package,
         new_lines
     )
-    with open(file, 'w') as f:
-        f.write(requirements)
+    if dry_run:
+        if verbose:
+            _verbose('Dry run, not editing ', file)
+        with open(file) as f:
+            old_requirements = f.read()
+        for line in difflib.unified_diff(old_requirements.splitlines(), requirements.splitlines()):
+            print(line)
```

Comment:
The correct way to print a nice diff is to do it like this:
```python
old = """
One
Two
Three
""".strip()

new = """
One
One and a half
Twoish
Three
""".strip()


print(
    "".join(
        difflib.unified_diff(
            old.splitlines(True), new.splitlines(True), fromfile="Old", tofile="New"
        )
    )
)
```

Then you get an output which resembles what you get with `git diff` kinda:
```
--- Old
+++ New
@@ -1,3 +1,4 @@
 One
-Two
+One and a half
+Twoish
 Three
```

This snippet works in Python 2 and Python 3.

Reply:
Thanks for all this help in fixing the MR!